src/main/MQ2Spawns.cpp

```diff
@@ -81,195 +72,464 @@ static bool dataNamingSpawn(const char* szIndex, MQTypeVar& Ret)
+		char value[24];
+		if (GetPrivateProfileString("Caption Colors", szName, "", value, sizeof(value), mq::internal_paths::MQini))
+		{
+			Enabled = GetBoolFromString(value, false);
+		}
+
+		if (!ToggleOnly)
+		{
+			char key[64];
+			sprintf_s(key, "%s-Color", szName);
+
+			if (GetPrivateProfileString("Caption Colors", key, "", value, sizeof(value), mq::internal_paths::MQini))
+			{
+				if (!sscanf_s(value, "%x", &Color.ARGB))
+					Color = DefaultColor;
+			}
+		}
 	}
-	else if (!_stricmp(Arg1, "Merc"))
+};
+
+static MQCaptionColor CaptionColors[] =
+{
+	// name          description                              defaultEnabled  toggle    defaultColor
+	{ "PC",          "Default color for PCs",                 false,          false,    0x003D68DC, },
+	{ "PCCon",       "PCs by con color",                      false,          true,     MQColor(),  },
+	{ "PCRaid",      "Raid members",                          false,          false,    0x0000FF7F, },
+	{ "PCClass",     "PCs by class color (raid settings)",    false,          true,     MQColor(),  },
+	{ "PCGroup",     "Group members",                         false,          false,    0x00FFFF00, },
+	{ "PCTrader",    "Traders",                               true,           false,    0x00FF7F00, },
+	{ "NPC",         "NPC default color",                     false,          false,    0x00FF0000, },
+	{ "NPCCon",      "NPCs by con color",                     true,           true,     MQColor(),  },
+	{ "NPCClass",    "NPCs by class color (raid settings)",   false,          true,     MQColor(),  },
+	{ "NPCMerchant", "NPC Merchants",                         true,           false,    0x00FF7F00, },
+	{ "NPCBanker",   "NPC Bankers",                           true,           false,    0x00C800FF, },
+	{ "NPCAssist",   "NPCs from main assist",                 true,           false,    0x00FFFF00, },
+	{ "NPCMark",     "Marked NPCs",                           true,           false,    0x00FFFF00, },
+	{ "PetNPC",      "Pet with NPC owner",                    false,          false,    0x00FF0000, },
+	{ "PetPC",       "Pet with PC owner",                     false,          false,    0x0017B8AD, },
+	{ "PetConColor", "Pet by con color",                      false,          true,     MQColor(),  },
+	{ "PetClass",    "Pet by class color (raid settings)",    false,          true,     MQColor(),  },
+	{ "Mercenary",   "Mercenaries",                           false,          false,    0x00FFB511, },
+	{ "Corpse",      "Corpses",                               false,          false,    0x00FF0000, },
+	{ "CorpseClass", "Corpse by class color (raid settings)", false,          true,     MQColor(),  },
+};
+
+
+static void CaptionCmd(PlayerClient* pChar, char* szLine)
+{
+	char Arg1[MAX_STRING] = { 0 };
+	GetArg(Arg1, szLine, 1);
+
+	if (!Arg1[0])
 	{
-		pCaption = gszSpawnMercName;
+		SyntaxError("Usage: /caption <settings|list|type <value>|update #|MQCaptions <on|off>>");
+		return;
```

Comment:
Just funny, I prefer going the other direction when dealing with else after return.  (leaving the else and removing the returns).